File: unibakonto/src/main/java/sk/pluk64/unibakonto/CardInfo.java
package sk.pluk64.unibakonto;

public class CardInfo {
    public final String number;
    private final String released;
    private final String validFrom;
    public final String validUntil;

    CardInfo(String number, String released, String validFrom, String validUntil) {
        this.number = divideBy4Digits(number);
        this.released = released;
        this.validFrom = validFrom;
        this.validUntil = validUntil;
    }

    private String divideBy4Digits(String number) {
        StringBuilder resultBuilder = new StringBuilder();

        int BLOCK_LENGTH = 4;
        int length = number.length();
        int firstBlockLength = length % BLOCK_LENGTH;
        resultBuilder.append(number, 0, firstBlockLength);

        for (int i = firstBlockLength; i < length; i += BLOCK_LENGTH) {
            if (i > 0) {
                resultBuilder.append(" ");
            }
            resultBuilder.append(number, i, i + BLOCK_LENGTH);
        }

        return resultBuilder.toString();
    }
}

File: unibakonto/src/main/java/sk/pluk64/unibakonto/CardInfo.kt
package sk.pluk64.unibakonto

import kotlinx.serialization.Serializable

@Serializable
data class CardInfo internal constructor(val number: String, val released: String, val validFrom: String, val validUntil: String)


fun String.divideBy4Digits(): String {
    val resultBuilder = StringBuilder()

    val blockLength = 4
    val length = length
    val firstBlockLength = length % blockLength
    resultBuilder.append(this, 0, firstBlockLength)

    var i = firstBlockLength
    while (i < length) {
        if (i > 0) {
            resultBuilder.append(" ")
        }
        resultBuilder.append(this, i, i + blockLength)
        i += blockLength
    }

    return resultBuilder.toString()
}
